vision/src/main/java/coppercore/vision/VisionLocalizer.java

```diff
@@ -1,62 +1,176 @@
+        for (int i = 0; i < inputs.length; i++) {
+            inputs[i] = new VisionIOInputsAutoLogged();
+        }
 
-    /**
-     * Constructs a VisionLocalizer with a given camera container.
-     *
-     * @param container The container holding cameras for localization.
-     */
-    public VisionLocalizer(CameraContainer container) {
-        this.container = container;
+        // Initialize disconnected alerts
+        this.disconnectedAlerts = new Alert[io.length];
+        for (int i = 0; i < inputs.length; i++) {
+            disconnectedAlerts[i] =
+                    new Alert("Vision camera " + i + " is disconnected.", AlertType.kWarning);
+        }
     }
 
     /** Periodically updates the camera data and processes new measurements. */
     @Override
     public void periodic() {
-        container.update();
-        for (Camera camera : container.getCameras()) {
-            if (camera.hasNewMeasurement()) {
-                cameraConsumer.accept(camera.getLatestMeasurement());
+        for (int i = 0; i < io.length; i++) {
+            io[i].updateInputs(inputs[i]);
+            Logger.processInputs("Vision/Camera" + i, inputs[i]);
+        }
+
+        // Initialize logging values
+        List<Pose3d> allRobotPoses = new LinkedList<>();
+        List<Pose3d> allRobotPosesAccepted = new LinkedList<>();
+        List<Pose3d> allRobotPosesRejected = new LinkedList<>();
+
+        // Loop over cameras
+        for (int cameraIndex = 0; cameraIndex < io.length; cameraIndex++) {
+            // Update disconnected alert
+            disconnectedAlerts[cameraIndex].set(!inputs[cameraIndex].connected);
+
+            // Initialize logging values
+            List<Pose3d> robotPoses = new LinkedList<>();
+            List<Pose3d> robotPosesAccepted = new LinkedList<>();
+            List<Pose3d> robotPosesRejected = new LinkedList<>();
+
+            for (VisionIO.PoseObservation observation : inputs[cameraIndex].poseObservations) {
+                if (rejectPose(observation)) {
+                    robotPosesRejected.add(observation.pose());
+                    continue;
+                }
+
+                robotPosesAccepted.add(observation.pose());
+
+                consumer.accept(
+                        observation.pose().toPose2d(),
+                        observation.timestamp(),
+                        getLatestVariance(observation, cameraIndex));
             }
+            logCameraData(cameraIndex, robotPoses, robotPosesAccepted, robotPosesRejected);
+
+            allRobotPoses.addAll(robotPoses);
+            allRobotPosesAccepted.addAll(robotPosesAccepted);
+            allRobotPosesRejected.addAll(robotPosesRejected);
+        }
+
+        logSummaryData(allRobotPoses, allRobotPosesAccepted, allRobotPosesRejected);
+    }
+
+    public void setVisionConsumer(VisionConsumer consumer) {
+        this.consumer = consumer;
+    }
+
+    private boolean rejectPose(VisionIO.PoseObservation observation) {
```

Comment:
Since this is a function that returns a boolean, let's name it `shouldRejectPose` or something similar